File: Backend/Backend/Models/Song.cs
using Newtonsoft.Json;

namespace Backend.Models
{
    public class Song
    {
        public string FileName { get; set; }
        public DateTime CreationDate { get; set; }
        public string Album { get; set; }
        public string Title { get; set; }
        public string Author { get; set; }
        public string Genre { get; set; }
        public int Likes { get; set; }
        public int Duration { get; set; }
        [JsonProperty("imageUrl")]
        public string ImageURL { get; set; }
        public List<string> UsersWhoLiked { get; set; } = new List<string>();
    }
}

File: Frontend/src/app/Interfaces/Song.ts
export interface Song
{
  title: string,
  author: string,
  album: string,
  creationDate: Date,
  duration: number,
  imageURL: string,
  fileName: string;
  likes: number,
  usersWhoLiked: string[],
  isLiked: boolean,
  isAddedToPlaylist: boolean;
}